wayback-core/src/main/java/org/archive/wayback/authenticationcontrol/IPMatchesBooleanOperator.java

```diff
@@ -62,15 +65,69 @@ public void setAllowedRanges(List<String> allowedRanges) {
 			}
 		}
 	}
+	
+	/**		
+	 * @return null. this is a placeholder for Spring's getter/setter 		
+	 * 			examination		
+	 */
+	public List<String> getTrustedProxies() {
+		return null;
+	}
+
+	/**
+	 * @param trustedProxies parses each String IPRange provided for the proxies, adding them to
+	 * 		the list of IPRanges which must be ignored by the IP match operator
+	 */
+	public void setTrustedProxies(List<String> trustedProxies) {
+		this.trustedProxies = new ArrayList<IPRange>();
+		for (String ip : trustedProxies) {
+			IPRange range = new IPRange();
+			if (range.setRange(ip)) {
+				this.trustedProxies.add(range);
+			} else {
+				LOGGER.severe("Unable to parse range (" + ip + ")");
+			}
+		}
+	}
+
+
+	public String getClientIPFromForwardedForHeader(String forwardedForHeader){
+		ArrayList<String> forwardingIPs;
+		String ip = null;
+		Boolean containsIP = false;
+		if (forwardedForHeader.contains(",")) {
+			forwardingIPs = new ArrayList<String>(Arrays.asList(forwardedForHeader.replace(" ", "").split(",")));
+			Collections.reverse(forwardingIPs);
+			for (String forwardingIP : forwardingIPs){
+				for (IPRange range : trustedProxies) {
+					if (range.contains(forwardingIP)) {
+						containsIP = true;
+						break;
+					}
+				}
+				if (!containsIP ||
+					(containsIP && forwardingIPs.get(forwardingIPs.size() -1).equals(forwardingIP))) {
```

Comment:
Thanks, @ldko! That makes sense. So do we need to add another constant in `WaybackRequest.java` to hold the remote address? Because the `REQUEST_REMOTE_ADDRESS` only holds the actual remote address when there is not a `X-Forwarded-For` value. 
https://github.com/iipc/openwayback/blob/master/wayback-core/src/main/java/org/archive/wayback/core/WaybackRequest.java#L1116

And the `isTrue` method of `IPMatchesBooleanOperator` in this PR sends the result of `WaybackRequest`'s `getRemoteIPAddress` to `getClientIPFromForwardedForHeader`--so that's either the remote address or the `X-Forwarded-For` value.